### PR TITLE
tests: Skip test for 14a004dc if all offsets are valid

### LINKED ISSUE #1781: VkLayerTest.VALIDATION_ERROR_01412 fails on AMD

### PR DESCRIPTION
**Pull request by cdwfs**
Also fixes some unexpected VU errors in this test that were
exposed when the main test accidentally passed. Fixes #1781.

### CODE REVIEW

**Comment by Tony-LunarG** [comment]
This looks like a good candidate for the device_profile_api layer (see ExceedMemoryAllocationCount for a lone example).  This looks good for now, but we may switch it over to use the layer at some point.

**Comment by cdwfs** [comment]
I fixed a few additional VU errors, and took a shot at using LUNARG_device_profile_api to artificially lower the offset limit. Seems to work on my end, but please let me know if I'm using the layer incorrectly.

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -8979,7 +8979,28 @@ TEST_F(VkLayerTest, VALIDATION_ERROR_14a004dc) {
...
+        // Attempt to artificially lower maximum offset
+        PFN_vkSetPhysicalDeviceLimitsEXT fpvkSetPhysicalDeviceLimitsEXT =
+            (PFN_vkSetPhysicalDeviceLimitsEXT)vkGetInstanceProcAddr(instance(), "vkSetPhysicalDeviceLimitsEXT");
+        if (!fpvkSetPhysicalDeviceLimitsEXT) {
+            printf("             All offsets are valid & device_profile_api not found; skipped.\n");
+            return;
+        }
+        default_props.limits.maxVertexInputAttributeOffset = default_props.limits.maxVertexInputBindingStride - 2;
+        fpvkSetPhysicalDeviceLimitsEXT(gpu(), &default_props.limits);
+        maxVertexInputAttributeOffset = default_props.limits.maxVertexInputAttributeOffset;
```
Comment (chrisforbes) [inline tests/layer_validation_tests.cpp]: Not a fan of this being called 'default_props' when you mangle it.

Comment (cdwfs) [inline tests/layer_validation_tests.cpp]: Renamed, & moved it out of scope when it's no longer in use, to prevent confusion with the final Device's props.

**Pull request merged**